presto-native-execution/presto_cpp/main/TaskManager.h

```diff
@@ -163,6 +163,10 @@ class TaskManager {
       "concurrent_lifespans_per_task"};
   static constexpr folly::StringPiece kSessionTimezone{"session_timezone"};
 
+  // This is called during shutting down. We terminate all running tasks which
+  // haven't been pinged by coordinator for a considerable time.
```

Comment:
```
nit: haven't received heartbeats from coordinator ...
```

Reply:
@xiaoxmeng 
> ```
> nit: haven't received heartbeats from coordinator ...
> ```

Actually, this is not what's happening.
We don't get heartbeats from the coordinator.
Coord asks as all sorts of things about the task and we update our what we call 'heartbeat' time.
So, then it boils down to 'coordinator forgot about the task' event.